drivers/sensor/max17055/max17055.c

```diff
@@ -0,0 +1,245 @@
+		tmp = capacity_to_ma(config->rsense_mohms,
+				     max17055->full_charge_capacity);
+		set_millis(val, tmp);
+		break;
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		tmp = capacity_to_ma(config->rsense_mohms,
+				     max17055->remaining_charge_capacity);
+		set_millis(val, tmp);
+		break;
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		/* Get time in ms */
+		if (max17055->time_to_empty == 0xffff) {
+			val->val1 = 0;
+			val->val2 = 0;
+		} else {
+			tmp = max17055->time_to_empty * 5625;
+			set_millis(val, tmp);
+		}
+		break;
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		if (max17055->time_to_full == 0xffff) {
+			val->val1 = 0;
+			val->val2 = 0;
+		} else {
+			/* Get time in ms */
+			tmp = max17055->time_to_full * 5625;
+			set_millis(val, tmp);
+		}
+		break;
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		val->val1 = max17055->cycle_count / 100;
+		val->val2 = max17055->cycle_count % 100 * 10000;
+		break;
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		tmp = capacity_to_ma(config->rsense_mohms,
+				     max17055->design_capacity);
+		set_millis(val, tmp);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIGN_VOLTAGE:
+		set_millis(val, config->design_voltage);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIRED_VOLTAGE:
+		set_millis(val, config->desired_voltage);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIRED_CHARGING_CURRENT:
+		val->val1 = config->desired_charging_current;
+		val->val2 = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int max17055_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct max17055_data *max17055 = dev->driver_data;
+	int status = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	status = max17055_reg_read(max17055, VCELL, &max17055->voltage);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, AVG_CURRENT,
+				   &max17055->avg_current);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, REP_SOC,
+				   &max17055->state_of_charge);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, INT_TEMP,
+				   &max17055->internal_temperature);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, REP_CAP,
+				   &max17055->remaining_charge_capacity);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, FULL_CAP_REP,
+				   &max17055->full_charge_capacity);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, TTE, &max17055->time_to_empty);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, TTF, &max17055->time_to_full);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, CYCLES, &max17055->cycle_count);
+	if (status) {
+		goto err;
+	}
+	status = max17055_reg_read(max17055, DESIGN_CAP,
+				   &max17055->design_capacity);
+	if (status) {
+		goto err;
+	}
+
+	return 0;
+err:
+	LOG_ERR("Failed to read internal temperature");
+	return -EIO;
+}
+
+/**
+ * @brief initialise the fuel gauge
+ *
+ * @return 0 for success
```

Comment:
Function returns `-EINVAL` also.. Should be mentioned here..

Reply:
done